spec/models/cloud_volume_spec.rb

```diff
@@ -6,4 +6,36 @@
 
     expect(described_class.available).to eq([cv2])
   end
+
+  context ".class_by_ems" do
+    let(:openstack_cloud) { FactoryGirl.create(:ems_openstack) }
+    let(:cinder) { FactoryGirl.create(:ems_cinder, :parent_ems_id => openstack_cloud.id) }
+    let(:amazon_cloud) { FactoryGirl.create(:ems_amazon) }
+    let(:ebs) { FactoryGirl.create(:ems_amazon_ebs, :parent_ems_id => amazon_cloud.id) }
+    let(:block_ems) { FactoryGirl.create(:ems_cinder) }
+
+    it "OpenStack cloud manager should return it's CloudVolume type" do
+      expect(CloudVolume.class_by_ems(openstack_cloud)).to eq(ManageIQ::Providers::Openstack::CloudManager::CloudVolume)
+    end
+
+    it "Cinder block storage manager should return CloudVolume type of the parent manager" do
+      expect(CloudVolume.class_by_ems(cinder)).to eq(ManageIQ::Providers::Openstack::CloudManager::CloudVolume)
+    end
+
+    it "Amazon cloud manager should return base CloudVolume type" do
```

Comment:
We should probably have these on Amazon side, we are trying not to mention any pluggable providers inside of the core code.